src/semgrep_agent/findings.py

```diff
@@ -99,9 +116,22 @@ def to_dict(self) -> Mapping[str, Any]:
 
 @dataclass(frozen=True)
 class FindingSets:
-    baseline: Set[Finding] = field(default_factory=set)
-    current: Set[Finding] = field(default_factory=set)
+    baseline_map: Dict[FindingKey, List[Finding]] = field(default_factory=dict)
+    current_map: Dict[FindingKey, List[Finding]] = field(default_factory=dict)
+
+    def expensive_new(self) -> Set[Finding]:
+        return self.current_set() - self.baseline_set()
+
+    @staticmethod
+    def _map_to_set(mapping: Mapping[FindingKey, List[Finding]]) -> Set[Finding]:
+        return set(
+            attr.evolve(f, index=ix)
+            for ff in mapping.values()
+            for ix, f in enumerate(ff)
+        )
```

Comment:
```python
        return set(
            attr.evolve(finding, index=index)
            for finding_map in mapping.values()
            for index, finding in enumerate(finding_map)
        )
```

Wrote this out for myself to understand what's going on, might as well update the code as well 🙃

Reply:
aye aye